File: web-admin/app/types/lead_capture/lead.ts
export interface Lead {
  lead_uuid: string;
  tenant_uuid: string;
  display_name?: string;
  phone?: string;
  email?: string;
  status: string;
  owner_user_uuid?: string;
  source_channel?: string;
  source_app_type?: string;
  source_account_uuid?: string;
  created_at?: string;
  updated_at?: string;
}

export interface LeadCreatePayload {
  display_name?: string;
  phone?: string;
  email?: string;
  source_channel?: string;
  source_app_type?: string;
  source_account_uuid?: string;
  owner_user_uuid?: string;
}

export interface LeadAssignment {
  assignment_uuid: string;
  tenant_uuid: string;
  lead_uuid: string;
  owner_user_uuid: string;
  reason?: string;
  created_at: string;
}

export interface LeadStatusHistory {
  history_uuid: string;
  tenant_uuid: string;
  lead_uuid: string;
  from_status: string;
  to_status: string;
  changed_at: string;
}

File: backend/internal/dto/lead_capture/lead.go
package lead_capture

// LeadCreateRequest defines payload for creating a lead.
type LeadCreateRequest struct {
	DisplayName       string `json:"display_name" binding:"omitempty" validate:"omitempty"`
	Phone             string `json:"phone" binding:"omitempty" validate:"omitempty"`
	Email             string `json:"email" binding:"omitempty" validate:"omitempty"`
	SourceChannel     string `json:"source_channel" binding:"omitempty" validate:"omitempty"`
	SourceAppType     string `json:"source_app_type" binding:"omitempty" validate:"omitempty"`
	SourceAccountUUID string `json:"source_account_uuid" binding:"omitempty" validate:"omitempty"`
	OwnerUserUUID     string `json:"owner_user_uuid" binding:"omitempty" validate:"omitempty"`
}

// LeadUpdateRequest defines payload for updating a lead.
type LeadUpdateRequest struct {
	DisplayName   string `json:"display_name" binding:"omitempty" validate:"omitempty"`
	Phone         string `json:"phone" binding:"omitempty" validate:"omitempty"`
	Email         string `json:"email" binding:"omitempty" validate:"omitempty"`
	Status        string `json:"status" binding:"omitempty" validate:"omitempty"`
	OwnerUserUUID string `json:"owner_user_uuid" binding:"omitempty" validate:"omitempty"`
}

// LeadListRequest defines list query for leads.
type LeadListRequest struct {
	Page          int    `form:"page" json:"page" binding:"omitempty" validate:"omitempty,min=1"`
	PageSize      int    `form:"page_size" json:"page_size" binding:"omitempty" validate:"omitempty,min=1,max=200"`
	Query         string `form:"q" json:"q" binding:"omitempty" validate:"omitempty"`
	Status        string `form:"status" json:"status" binding:"omitempty" validate:"omitempty"`
	OwnerUserUUID string `form:"owner_user_uuid" json:"owner_user_uuid" binding:"omitempty" validate:"omitempty"`
	ChannelCode   string `form:"channel_code" json:"channel_code" binding:"omitempty" validate:"omitempty"`
}
